PusherMainView/PusherMainView/FCMPusher.swift

```diff
@@ -0,0 +1,183 @@
+                                                userInfo: [NSLocalizedDescriptionKey: "Unknown error"])))
+                }
+                return
+            }
+
+            switch r.statusCode {
+            case 200:
+                DispatchQueue.main.async {
+                    completion(.success(HTTPURLResponse.localizedString(forStatusCode: r.statusCode)))
+                }
+
+            default:
+                if let data = data,
+                   let html = String(data: data, encoding: .utf8) {
+                    DispatchQueue.main.async {
+                        completion(.failure(NSError(domain: "com.pusher.FCMPusher",
+                                                    code: r.statusCode,
+                                                    userInfo: [NSLocalizedDescriptionKey: html.slice(between: "<TITLE>", and: "</TITLE>")])))
+                    }
+
+                } else {
+                    DispatchQueue.main.async {
+                        let error = NSError(domain: "com.pusher.FCMPusher",
+                                            code: r.statusCode,
+                                            userInfo: [NSLocalizedDescriptionKey: HTTPURLResponse.localizedString(forStatusCode: r.statusCode)])
+                        completion(.failure(error))
+                    }
+                }
+            }
+        }.resume()
+    }
+
+    func pushUsingV1Endpoint(_ token: String,
+                             payload: [String: Any],
+                             collapseID: String?,
+                             serverKey: String,
+                             projectID: String,
+                             completion: @escaping (Result<String, Error>) -> Void) {
+
+        guard let url = URL(string: "https://fcm.googleapis.com/v1/projects/\(projectID)/messages:send") else {
+            completion(.failure(NSError(domain: "com.pusher.FCMPusher",
+                                        code: 0,
+                                        userInfo: [NSLocalizedDescriptionKey: "URL error"])))
+            return
+        }
+
+        var updatedPayload = payload
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("Bearer \(serverKey)", forHTTPHeaderField: "Authorization")
+
+        updatedPayload["token"] = token
+        if let collapseID = collapseID {
+            updatedPayload["collapse_key"] = collapseID
+        }
+
+        guard let httpBody = try? JSONSerialization.data(withJSONObject: ["message": payload], options: .prettyPrinted) else {
+            completion(.failure(NSError(domain: "com.pusher.FCMPusher",
+                                        code: 0,
+                                        userInfo: [NSLocalizedDescriptionKey: "Payload error"])))
+            return
+        }
+
+        request.httpBody = httpBody
```

Comment:
@imaginaris Same comment, it can be more readable if you move this URLRequest creation to an extension / helper.